pandas/_libs/hashtable_class_helper.pxi.in

```diff
@@ -870,7 +870,7 @@ cdef class PyObjectHashTable(HashTable):
         for i in range(n):
             val = values[i]
             hash(val)
-            if not checknull(val):
+            if not checknull(val) or val is None:
```

Comment:
needs a comment here, checkull is specifically designed to catch ALL nulls here.